Source/Laser/LaserProfiles.H

```diff
@@ -179,6 +204,138 @@ private:
+    * \param np: number of laser particles
+    * \param Xp: pointer to first component of positions of laser particles
+    * \param Yp: pointer to second component of positions of laser particles
+    * \param t: Current physical time
+    * \param amplitude: pointer to array of field amplitude.
+    */
+    void internal_fill_amplitude_nonuniform(
+        const int idx_t_left,
+        const int np,
+        amrex::Real const * AMREX_RESTRICT const Xp,
+        amrex::Real const * AMREX_RESTRICT const Yp,
+        amrex::Real t,
+        amrex::Real * AMREX_RESTRICT const amplitude) const;
+
+private:
+    /** \brief parse a field file in the binary 'txye' format (whose details are given below).
+    *
+    * A 'txye' file should be a binary file with the following format:
+    * -flag to indicate if the grid is uniform or not (1 byte, 0 means non-uniform, !=0 means uniform)
+    * -np, number of timesteps (uint32_t, must be >=2)
+    * -nx, number of points along x (uint32_t, must be >=2)
+    * -ny, number of points along y (uint32_t, must be 1 for 2D simulations and >=2 for 3D simulations)
+    * -timesteps (double[2] if grid is uniform, double[np] otherwise)
+    * -x_coords (double[2] if grid is uniform, double[nx] otherwise)
+    * -y_coords (double[1] if 2D, double[2] if 3D & uniform grid, double[ny] if 3D & non uniform grid)
+    * -field_data (double[nt * nx * ny], with nt being the slowest coordinate).
+    * The spatiotemporal grid must be rectangular.
+    *
+    * \param txye_file_name: name of the file to parse
+    */
+    void parse_txye_file(std::string txye_file_name);
+
+    /** \brief Finds left and right time indices corresponding to time t
+    *
+    *
+    * \param t: simulation time
+    */
+    std::pair<int,int> find_left_right_time_indices(amrex::Real t) const;
+
+    /** \brief Load field data within the temporal range [t_begin, t_end)
+    *
+    * Must be called after having parsed a data file with parse_txye_file.
+    *
+    * \param t_begin: left limit of the timestep range to read
+    * \param t_end: right limit of the timestep range to read (t_end is not read)
+    */
+    void read_data_t_chuck(int t_begin, int t_end);
+
+    struct{
+        std::string txye_file_name; //Name of the file containing the data
+        bool is_grid_uniform = false; //Flag to store if the grid is uniform
+        int nt, nx, ny; //Dimensions of E_data
+        amrex::Gpu::ManagedVector<amrex::Real> t_coords; //Vector of temporal coordinates
+        amrex::Gpu::ManagedVector<amrex::Real> x_coords; //Vector of x coordinates
+        amrex::Gpu::ManagedVector<amrex::Real> y_coords; //Vector of y coordinates
+        int time_chunk_size; //Size of the timestep range to load
+        int first_time_index; //Index of the first timestep in memory
+        int last_time_index; //Index of the last timestep in memory
+        amrex::Gpu::ManagedVector<amrex::Real> E_data; //Field data
+    } m_params;
```

Comment:
I am fine with keeping the code as is, and add documentation to avoid confusion.
Also, let's make sure the documentation can be read by Doxygen (as an example the two links below show the Doxygen-generated documentation for class ILaserProfile on `dev` branch, let's make sure Doxygen also reads members of `m_params`).
https://warpx.readthedocs.io/en/latest/_static/doxyhtml/class_warp_x_laser_profiles_1_1_i_laser_profile.html
https://warpx.readthedocs.io/en/latest/api/class_warp_x_laser_profiles_1_1_i_laser_profile.html

Reply:
Wow, the documentation is awesome!

Reply:
I've just added the comments...